### ISSUE TITLE
bst workspace open --force doesnt work

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#364]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/364)
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 17, 2018, 08:11

When I ran: `bst workspace open --force core/glib-networking.bst ~/GLIB_NETWORKING` and the target directory was not empty, I got the following results:

```
[--:--:--][][] START   Loading pipeline
[00:00:01][][] SUCCESS Loading pipeline
[--:--:--][][] START   Resolving pipeline
[00:00:01][][] SUCCESS Resolving pipeline
[--:--:--][][] START   Resolving cached state
[00:00:01][][] SUCCESS Resolving cached state
[--:--:--][][] START   Checking sources
[00:00:00][][] SUCCESS Checking sources
[--:--:--][][] START   Staging sources to /home/tristan/GLIB_NETWORKING
[--:--:--][        ][ main:core/glib-networking.bst-0    ] STATUS  Running host command

    /usr/bin/git show 9c03e2300eeeb2f335d5146328cd2eb8ca0b99e5:.gitmodules

[--:--:--][][] START   Setting up workspace "/home/tristan/GLIB_NETWORKING"
[00:00:00][][] FAILURE Setting up workspace "/home/tristan/GLIB_NETWORKING"
[00:00:00][][] FAILURE Staging sources to /home/tristan/GLIB_NETWORKING

git source at core/glib-networking.bst [line 3 column 2]: Failed to clone git mirror /home/tristan/.cache/buildstream/sources/git/https___git_gnome_org_browse_glib_networking in directory: /home/tristan/GLIB_NETWORKING/
```

Probably we need to stage the sources into an empty temp directory, and use `utils.link_files()` from the temp directory instead of giving the Source implementation a non-empty directory and hoping for the best.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 17, 2018, 09:11

mentioned in commit 5741300db0503fca148f81bb83153dbd0dc5c77e

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 17, 2018, 09:11

mentioned in commit 248877974f6cf68023323a070c969f31792fb804

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 17, 2018, 09:32

mentioned in commit 098ee3cf6c742439cd48534d2a0ded6c2c1138fb

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 17, 2018, 09:32

mentioned in commit 4fc2da9ad0ac077db7b675db272d901d2249f66f

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 17, 2018, 09:46

closed via merge request !409

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 17, 2018, 09:46

closed via commit 4fc2da9ad0ac077db7b675db272d901d2249f66f
